cpp/src/arrow/util/iterator.h

```diff
@@ -88,10 +90,51 @@ class Iterator {
   }
 
   /// Iterators will only compare equal if they are both null
-  bool operator==(const Iterator& other) const { return ptr_ == other.ptr_; }
+  bool Equals(const Iterator& other) const { return ptr_ == other.ptr_; }
```

Comment:
Could you add explanation that this is required to check the `end()` condition? It wasn't obvious when I first looked.

Reply:
Will do